pkg/util/patch/patch.go

```diff
@@ -26,30 +26,29 @@ type patcher struct {
 }
 
 // Create is a helper that creates a patch out of the change made in applyFn
-func (p *patcher) Create(new runtime.Object, applyFn func(runtime.Object) error) ([]byte, error) {
+func (p *patcher) Create(new runtime.Object, applyFn func(runtime.Object) error) (patchBytes []byte, err error) {
+	var oldBytes, newBytes bytes.Buffer
+	encoder := p.serializer.Encoder()
 	old := new.DeepCopyObject().(runtime.Object)
-	encoder := p.serializer.Encoder(serializer.ContentTypeJSON)
 
-	oldbytes, err := encoder.Encode(old)
-	if err != nil {
-		return nil, err
+	if err = encoder.Encode(serializer.NewJSONFrameWriter(&oldBytes), old); err != nil {
```

Comment:
super minor nit: I prefer `if err := foo(); err != nil {}`, specifically the `:=` part generally, but I see you're using named returns here so it's fine